### PR TITLE
#54 pytest-extension

### LINKED ISSUE #54: Add new plugin - pytest-extension
Initially provide the following fixtures:

- database connection based on `backend_aware_database_params`
- bucket-fs Pathlike object based on `backend_aware_bucketfs_params`
- a function creating a bucket-fs connection object

### PR DESCRIPTION
**Pull request by ahsimb**
closes #54

### CODE REVIEW

File: pytest-extension/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# pytest-exasol-extension Plugin
+
+The `pytest-exasol-extension` plugin provides a pytest fixtures for preparing a database for the extension tests.
```
Comment (ckunki) [inline pytest-extension/README.md:3]:
```suggestion
The `pytest-exasol-extension` plugin provides pytest fixtures for preparing a database for the extension tests.
```

File: pytest-extension/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Usage in Tests
+
+Below is an example of a test that requires a database connection with an open test schema.
+
+```python
+import pytest
+
+@pytest.fixture(scope="session")
+def db_schema_name() -> str:
```
Comment (ckunki) [inline pytest-extension/README.md:23]: This fixture is not used directly by the tests below. 
Is the fixture required by pyexasol_connection or by the backend plugin? 
If so, then please add a comment, e.g. "This fixture is required by ...".

File: pytest-extension/exasol/pytest_extension/__init__.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+def pyexasol_connection(backend_aware_database_params,
+                        db_schema_name
+                        ) -> pyexasol.ExaConnection:
+    """
+    The fixture provides a database connection. It opens the test schema,
+    creating it if it doesn't exist. In the latter case the schema gets
+    deleted and the end of the fixture's life span.
+    """
+    with pyexasol.connect(**backend_aware_database_params, compression=True) as conn:
+        sql = f"SELECT * FROM SYS.EXA_SCHEMAS WHERE SCHEMA_NAME = '{db_schema_name}';"
```
Comment (ckunki) [inline pytest-extension/exasol/pytest_extension/__init__.py:75]: Should we remove the trailing semicolon `;` from the SQL statements?
AFAIK it's useless.

File: pytest-extension/exasol/pytest_extension/__init__.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        no_schema = len(conn.execute(sql).fetchall()) == 0
```
Comment (ckunki) [inline pytest-extension/exasol/pytest_extension/__init__.py:76]: Would `temp_schema` be a better name for this variable?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: pytest-extension/exasol/pytest_extension/__init__.py (hunk 3 above)
Comment (ahsimb) [reply pytest-extension/exasol/pytest_extension/__init__.py:75]: A semicolon separates statements in a script. With one statement it is not needed. The presence or absence of it is just a question of style. I don't have much of an option on that. Can remove it.

File: pytest-extension/exasol/pytest_extension/__init__.py (hunk 4 above)
Comment (ahsimb) [reply pytest-extension/exasol/pytest_extension/__init__.py:76]: `use_temp_schema`, for that matter. The var is a boolean indicator. `temp_schema` for me sounds like the name of a temporary schema.

**Review by ckunki — APPROVED** [review]

**Pull request merged**